src/components/include/cos_kernel_api.h

```diff
@@ -88,6 +89,8 @@ int cos_sched_rcv(arcvcap_t rcv, thdid_t *thdid, int *blocked, cycles_t *cycles)
 
 int cos_introspect(struct cos_compinfo *ci, capid_t cap, unsigned long op);
 
+int cos_sinv(sinvcap_t sinv);
```

Comment:
Shouldn't this also take 4 additional arguments that we can use to pass to the invocation?